src/allmydata/storage/http_server.py

```diff
@@ -137,31 +142,31 @@ def handle_route(*args, **kwargs):
     return decorator
 
 
-@attr.s
+@define
 class StorageIndexUploads(object):
     """
     In-progress upload to storage index.
     """
 
     # Map share number to BucketWriter
-    shares = attr.ib(factory=dict)  # type: Dict[int,BucketWriter]
+    shares: dict[int, BucketWriter] = field(factory=dict)
```

Comment:
I also recently learned that `... = Factory(dict)` works.

Reply:
Handy